docs/changelog.md

```diff
@@ -1,5 +1,9 @@
 dev
 
+- Now store metadata information for each application installed, including install options like `--spec`, and injected packages. This information allows upgrade, upgrade-all and reinstall-all to work properly even with non-pypi installed packages.  (#222)
+- Change `upgrade`, removing options `--spec`, `--include-deps`.  Use original options used to install application instead. (#222)
+- Change `upgrade-all`, removing options `--include-deps`, `--system-site-packages`, `--index-url`, `--editable`, `--pip-args`.  Use original options used to install each application instead. (#222)
+- Change `reinstall-all`, removing options `--include-deps`, `--system-site-packages`, `--index-url`, `--editable`, `--pip-args`.  Use original options used to install each application instead. (#222)
```

Comment:
nit: The way this is phased sometimes comes across like a user has to take action. For example 

> Now store metadata information

could be written as

> Metadata is now stored for each application...

And

> Use original options used to install each application instead.

might be more clear as 

> Original options are used to install each application instead

Reply:
Yeah you're right, I was in "git commit" mode when writing these.  I'll make them more passive.

Reply:
I fixed the ones I added in this PR, but I notice that some of the previous changelog entries sound similarly commanding.  I left them be for now.